x/mongo/driver/operation.go

```diff
@@ -202,6 +202,10 @@ type Operation struct {
 	// ServerAPI specifies options used to configure the API version sent to the server.
 	ServerAPI *ServerAPIOptions
 
+	// IsOutputAggregate specifies whether this operation is an aggregate with an output stage. If true,
+	// read preference will not be added to the command on wire versions < 13.
+	IsOutputAggregate bool
```

Comment:
Read-preference appension happens in this file (in `createReadPref`), but I needed to pass down whether or not the underlying operation is an aggregate with an output stage from `Collection#aggregate(...)`. I've added this extra field to `Operation` and a `HasOutputStage` to `Aggregate`, but this feels a little clunky. Let me know if you have any better ideas.